packages/cloud-functions/src/cico/xanpoolWebhook.ts

```diff
@@ -0,0 +1,61 @@
+import * as functions from 'firebase-functions'
+import { trackEvent } from '../bigQuery'
+import { saveTxHashProvider } from '../firebase'
+import { Providers } from './Providers'
+import { flattenObject } from './utils'
+
+const XANPOOL_BIG_QUERY_EVENT_TABLE = 'cico_provider_events_xanpool'
+
+interface XanpoolEventPayload {
+  timestamp: number
+  message: string
+  payload: {
+    id: string
+    type: string
+    status: string
+    method: string
+    crypto: number
+    fiat: number
+    total: number
+    currency: string
+    cryptoCurrency: string
+    serviceCharge: number
+    cryptoPrice: number
+    cryptoPriceUsd: number
+    nonce: string
+    qrCode: string
+    wallet?: string
+    userCountry: string
+    userId: string
+    peer: {
+      account: string
+    }
+    blockchainTxId?: string
+    depositWallets: {
+      celo: string
+    }
+  }
+}
+
+export const parseXanpoolEvent = async (reqBody: any) => {
+  const { timestamp, message, payload }: XanpoolEventPayload = reqBody
+  if (payload.blockchainTxId && payload.wallet) {
+    saveTxHashProvider(payload.wallet, payload.blockchainTxId, Providers.Xanpool)
+  }
+
+  const data = flattenObject({ timestamp, message, ...payload })
+  await trackEvent(XANPOOL_BIG_QUERY_EVENT_TABLE, data)
+}
+
+// Function needs to be all lowercase due to a bug in Xanpool's dashboard
+export const xanpoolwebhook = functions.https.onRequest(async (req, res) => {
+  try {
+    await parseXanpoolEvent(req.body)
```

Comment:
They have no signature or verification method for us to check, right? So even though it will most likely not happen someone could technically send fake xanpool requests to our server to mess with our stats.

Nothing to do, just checking

Reply:
Yup, unfortunately no protections in place. Has the potential to be spammed, though I don't see why anyone would want to. We've got a live dashboard now so we should be able to see if this becomes a problem and make adjustments if needed :)